src/sentry/incidents/subscription_processor.py

```diff
@@ -522,31 +608,73 @@ def process_update(self, subscription_update: QuerySubscriptionUpdate) -> None:
         with transaction.atomic(router.db_for_write(AlertRule)):
             # Triggers is the threshold - NOT an instance of a trigger
             for trigger in self.triggers:
-                if alert_operator(
-                    aggregation_value, trigger.alert_threshold
-                ) and not self.check_trigger_matches_status(trigger, TriggerStatus.ACTIVE):
-                    # If the value has breached our threshold (above/below)
-                    # And the trigger is not yet active
-                    metrics.incr("incidents.alert_rules.threshold", tags={"type": "alert"})
-                    # triggering a threshold will create an incident and set the status to active
-                    incident_trigger = self.trigger_alert_threshold(trigger, aggregation_value)
-                    if incident_trigger is not None:
-                        fired_incident_triggers.append(incident_trigger)
+                if self.has_feature_flag:
+                    if trigger.alert_rule.detection_type == AlertRuleDetectionType.DYNAMIC:
+                        # MF: There should only be one anomaly in the list
+                        for anomaly in anomalies:
+                            if self.get_anomaly_detected(
+                                anomaly
+                            ) and not self.check_trigger_matches_status(
+                                trigger, TriggerStatus.ACTIVE
+                            ):
+                                metrics.incr(
+                                    "incidents.alert_rules.threshold", tags={"type": "alert"}
+                                )
+                                incident_trigger = self.trigger_alert_threshold(
+                                    trigger, aggregation_value
+                                )
+                                if incident_trigger is not None:
+                                    fired_incident_triggers.append(incident_trigger)
+                            else:
+                                self.trigger_alert_counts[trigger.id] = 0
+
+                            # MF: There's no explicit "resolve threshold" for anomaly detection. I chose to resolve if no anomaly detected, but unsure if that's correct.
```

Comment:
This makes sense to me - we might change it in the future but for now it's fine.